code/__HELPERS/_lists.dm

```diff
@@ -473,17 +473,23 @@
  * You should only pass integers in.
  */
 /proc/pick_weight(list/list_to_pick)
+	if(!length(list_to_pick))
+		return null
+
 	var/total = 0
-	var/item
-	for(item in list_to_pick)
+	for(var/item in list_to_pick)
 		if(!list_to_pick[item])
 			list_to_pick[item] = 0
 		total += list_to_pick[item]
 
 	total = rand(1, total)
-	for(item in list_to_pick)
-		total -= list_to_pick[item]
-		if(total <= 0 && list_to_pick[item])
+	for(var/item in list_to_pick)
+		var/item_weight = list_to_pick[item]
+		if(!item_weight)
```

Comment:
```suggestion
		if(item_weight == 0)
```

i'm not certain on this PR as a whole but let's not treat integer values like a boolean

Reply:
But it's effectively the same. But I suppose there were codestyle changes, so you started avoiding type autocasting?

Reply:
a lot of old code keeps using implicit assumptions but i personally prefer to be more explicit and think it should be kept to `if(!BOOLEAN)` instead of `if(!INTEGER)`, people can do wacky stuff otherwise in order to make the code more confusing in later prs